soccer/planning/RRTPlanner.cpp

```diff
@@ -55,7 +107,28 @@ bool RRTPlanner::shouldReplan(const SinglePlanRequest& planRequest,
     return false;
 }
 
-const int maxContinue = 10;
+std::unique_ptr<Path> RRTPlanner::plan(
+    const MotionInstant start, const MotionInstant goal,
+    const MotionConstraints& motionConstraints,
+    const Geometry2d::ShapeSet& obstacles) {
+    // Run bi-directional RRT to generate a path.
+    auto points = runRRT(start, goal, motionConstraints, obstacles);
+
+    // Optimize out uneccesary waypoints
+    optimize(points, obstacles, motionConstraints, start.vel, goal.vel);
+
+    // Check if Planning or optimization failed
+    if (points.size() < 2) {
+        debugLog("PathPlanning Failed");
+        return nullptr;
+    }
+
+    // Generate and return a cubic bezier path using the waypoints
+    return generateCubicBezier(points, obstacles, motionConstraints, start.vel,
+                               goal.vel);
+}
+
+const int maxContinue = 60;
 
 std::unique_ptr<Path> RRTPlanner::run(SinglePlanRequest& planRequest) {
```

Comment:
This method could use a few comments. It's a little tough to follow what all's going on in all the if-statements and things.